### PR TITLE
Added external links support

### PR DESCRIPTION
**Pull request by cccs-sgaron**

### CODE REVIEW

File: assemblyline/odm/models/config.py [hunk 1]
```diff
@@ -999,6 +999,64 @@ class AlertingMeta(odm.Model):
 }
 
 
+@odm.model(index=False, store=False, description="Target definition of an external link")
+class ExternalLinksTargets(odm.Model):
+    type: str = odm.Enum(values=['metadata', 'tag', 'hash'], description="Type of external link target")
+    key: str = odm.Keyword(description="Key that it can be used againts")
```
Comment (cccs-kevin) [inline assemblyline/odm/models/config.py:1005]: against*

File: assemblyline/odm/models/config.py [hunk 2]
```diff
@@ -1049,9 +1107,11 @@ class UI(odm.Model):
     download_encoding = odm.Enum(values=["raw", "cart"], description="Which encoding will be used for downloads?")
     email: str = odm.Optional(odm.Email(), description="Assemblyline admins email address")
     enforce_quota: bool = odm.Boolean(description="Enforce the user's quotas?")
+    external_links: List[ExternalLinks] = odm.List(
+        odm.Compound(ExternalLinks),
+        description="List of external pivots links")
```
Comment (cccs-kevin) [inline assemblyline/odm/models/config.py:1112]: external pivot* links

**Review by cccs-kevin — APPROVED** [review]

File: assemblyline/odm/models/config.py [hunk 3]
```diff
@@ -999,6 +999,64 @@ class AlertingMeta(odm.Model):
...
+class ExternalLinksTargets(odm.Model):
+    type: str = odm.Enum(values=['metadata', 'tag', 'hash'], description="Type of external link target")
+    key: str = odm.Keyword(description="Key that it can be used againts")
+
+
+@odm.model(index=False, store=False, description="External links that specific metadata and tags can pivot to")
+class ExternalLinks(odm.Model):
+    allow_bypass: bool = odm.boolean(
+        default=False,
+        description="If the classification of the item is higher then the max_classificaiton, can we let the user "
```
Comment (cccs-ml) [inline assemblyline/odm/models/config.py:1012]: then -> than

File: assemblyline/odm/models/config.py [hunk 4]
```diff
@@ -999,6 +999,64 @@ class AlertingMeta(odm.Model):
...
+        {"type": "hash", "key": "sha256"},
+    ],
+    "url": "https://www.virustotal.com/gui/search/{REPLACE}",
+    "double_encode": True,
+    # "classification": "TLP:CLEAR",
+    # "max_classification": "TLP:CLEAR",
+}
+
+EXAMPLE_EXTERNAL_LINK_MB_SHA256 = {
+    # This is an example on how this would work with VirusTotal
```
Comment (cccs-ml) [inline assemblyline/odm/models/config.py:1048]: VirusTotal -> Malware Bazaar

**Review by cccs-ml — APPROVED** [review]

**Pull request merged**